recipes-core/tdx-enc-handler/tdx-enc-handler/tdx-enc-caam.sh

```diff
@@ -94,6 +121,20 @@ tdx_enc_partition_mount() {
     fi
 }
 
+# restore data if available
+tdx_enc_restore_data() {
+    if [ ${TDX_ENC_PRESERVE_DATA} -eq 1 ]; then
+        tdx_enc_log "Checking backup file: ${TDX_ENC_PRESERVE_LOCATION}"
+
+        if [ -f ${TDX_ENC_PRESERVE_LOCATION} ]; then
+            tdx_enc_log "Restoring original content..."
+
+            bzip2 -cd ${TDX_ENC_PRESERVE_LOCATION} | tar --strip-components 1 -C ${TDX_ENC_STORAGE_MOUNTPOINT} -xf -
```

Comment:
We shouldn't need the `--strip-components` parameter here since we know the files were packed relative to the current directory (`.`).

Reply:
In my local testing this seemd to be required as we strip into the storage location, however the backup also included this path. So what I was seeing is data restored into `/data/data` when it should just have been `/data`.
I expect the backup function should probably be changed to avoid this instead to be more precise?

Reply:
Latest update:
Apparently I was wrong. The image I was testing was an older one. After a new clean build the backup is now correct.
I can only assume it is because the latest code now does the `dd` command instead of the `tee` I originally had.

Reply:
@rborn-tx - Are you happy to resolve this one as the latest code has this change.